File: skills/api-design/examples/python/fastapi/crud_example.py
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel, EmailStr
from typing import List, Optional

app = FastAPI(title="User API")

users_db = []
next_id = 1

class UserCreate(BaseModel):
    name: str
    email: EmailStr

class User(BaseModel):
    id: int
    name: str
    email: str

@app.get("/users", response_model=dict)
def get_users(page: int = 1, limit: int = 20):
    return {
        "data": users_db,
        "pagination": {"page": page, "limit": limit, "total": len(users_db)}
    }

@app.get("/users/{user_id}", response_model=dict)
def get_user(user_id: int):
    user = next((u for u in users_db if u["id"] == user_id), None)
    if not user:
        raise HTTPException(status_code=404, detail="User not found")
    return {"data": user}

@app.post("/users", response_model=dict, status_code=201)
def create_user(user: UserCreate):
    global next_id
    new_user = {"id": next_id, "name": user.name, "email": user.email}
    users_db.append(new_user)
    next_id += 1
    return {"data": new_user}

@app.patch("/users/{user_id}", response_model=dict)
def update_user(user_id: int, user: UserCreate):
    for u in users_db:
        if u["id"] == user_id:
            u["name"] = user.name
            u["email"] = user.email
            return {"data": u}
    raise HTTPException(status_code=404, detail="User not found")

@app.delete("/users/{user_id}", status_code=204)
def delete_user(user_id: int):
    global users_db
    users_db = [u for u in users_db if u["id"] != user_id]

File: .claude/skills/api-design/examples/javascript/express/crud_example.js
const express = require('express');
const app = express();

app.use(express.json());

let users = [];
let nextId = 1;

app.get('/users', (req, res) => {
  const { page = 1, limit = 20 } = req.query;
  res.json({
    data: users,
    pagination: { page: Number(page), limit: Number(limit), total: users.length }
  });
});

app.get('/users/:id', (req, res) => {
  const user = users.find(u => u.id === parseInt(req.params.id));
  if (!user) return res.status(404).json({ error: 'User not found' });
  res.json({ data: user });
});

app.post('/users', (req, res) => {
  const newUser = { id: nextId++, ...req.body };
  users.push(newUser);
  res.status(201).json({ data: newUser });
});

app.patch('/users/:id', (req, res) => {
  const user = users.find(u => u.id === parseInt(req.params.id));
  if (!user) return res.status(404).json({ error: 'User not found' });
  Object.assign(user, req.body);
  res.json({ data: user });
});

app.delete('/users/:id', (req, res) => {
  users = users.filter(u => u.id !== parseInt(req.params.id));
  res.status(204).send();
});

app.listen(3000, () => console.log('API running on port 3000'));